dipy/workflows/reconst.py

```diff
@@ -1902,3 +1963,862 @@ def run(
+                min_separation_angle=min_separation_angle,
+                mask=mask_vol,
+                return_sh=True,
+                sh_order_max=sh_order_max,
+                normalize_peaks=True,
+                parallel=parallel,
+                num_processes=num_processes,
+            )
+            peaks_sfm.affine = affine
+
+            save_pam(opam, peaks_sfm)
+
+            logging.info("SFM computation completed.")
+
+            if extract_pam_values:
+                pam_to_niftis(
+                    peaks_sfm,
+                    fname_shm=oshm,
+                    fname_peaks_dir=opeaks_dir,
+                    fname_peaks_values=opeaks_values,
+                    fname_peaks_indices=opeaks_indices,
+                    fname_gfa=ogfa,
+                    fname_sphere=osphere,
+                    fname_b=ob,
+                    fname_qa=oqa,
+                    reshape_dirs=True,
+                )
+
+            dname_ = os.path.dirname(opam)
+            msg = (
+                "Pam5 file saved in current directory"
+                if dname_ == ""
+                else f"Pam5 file saved in {dname_}"
+            )
+            logging.info(msg)
+
+            return io_it
+
+
+class ReconstGQIFlow(Workflow):
+    @classmethod
+    def get_short_name(cls):
+        return "gqi"
+
+    def run(
+        self,
+        input_files,
+        bvalues_files,
+        bvectors_files,
+        mask_files,
+        *,
+        method="gqi2",
+        sampling_length=1.2,
+        normalize_peaks=False,
+        sphere_name=None,
+        b0_threshold=50.0,
+        bvecs_tol=0.01,
+        sh_order_max=8,
+        relative_peak_threshold=0.5,
+        min_separation_angle=25,
+        parallel=False,
+        extract_pam_values=False,
+        num_processes=None,
+        out_dir="",
+        out_pam="peaks.pam5",
+        out_shm="shm.nii.gz",
+        out_peaks_dir="peaks_dirs.nii.gz",
+        out_peaks_values="peaks_values.nii.gz",
+        out_peaks_indices="peaks_indices.nii.gz",
+        out_gfa="gfa.nii.gz",
+        out_sphere="sphere.txt",
+        out_b="B.nii.gz",
+        out_qa="qa.nii.gz",
+    ):
+        """Workflow for Generalized Q-Sampling Imaging (GQI)
+
+        See :footcite:p:`Yeh2010` for further details about the method.
+
+        Parameters
+        ----------
+        input_files : string
+            Path to the input volumes. This path may contain wildcards to
+            process multiple inputs at once.
+        bvalues_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        bvectors_files : string
+            Path to the bvalues files. This path may contain wildcards to use
+            multiple bvalues files at once.
+        mask_files : string
+            Path to the input masks. This path may contain wildcards to use
+            multiple masks at once. (default: No mask used)
```

Comment:
@skoudoro It says default No mask used but there is no way to leave it empty from the command line.

Reply:
Good catch, removing